src/sentry/incidents/endpoints/organization_alert_rule_index.py

```diff
@@ -177,14 +177,22 @@ def get(self, request: Request, organization) -> Response:
                 team_filter_query = team_filter_query | Q(owner_id=None)
 
         alert_rules = AlertRule.objects.fetch_for_organization(organization, projects)
-        if not features.has("organizations:performance-view", organization):
-            # Filter to only error alert rules
-            alert_rules = alert_rules.filter(snuba_query__dataset=Dataset.Events.value)
         issue_rules = Rule.objects.filter(
             status__in=[ObjectStatus.ACTIVE, ObjectStatus.DISABLED],
             source__in=[RuleSource.ISSUE],
             project__in=projects,
         )
+
+        if not features.has("organizations:performance-view", organization):
+            # Filter to only error alert rules
+            alert_rules = alert_rules.filter(snuba_query__dataset=Dataset.Events.value)
+        else:
+            # Filter only non-issue alert rules
+            datasets = request.GET.getlist("dataset", [])
+            if len(datasets) > 0:
+                alert_rules = alert_rules.filter(snuba_query__dataset__in=datasets)
+                issue_rules = issue_rules.filter(source=-1)
```

Comment:
Are you doing this on purpose to invalidate the query that is being built?

Reply:
I replaced it with `Rule.objects.none()`. Thanks for the suggestion!